arex-storage-web-api/src/main/java/com/arextest/storage/web/controller/AgentRecordingController.java

```diff
@@ -113,21 +153,73 @@ public Response save(@RequestBody AREXMocker requestType) {
     }
     try {
       MDCTracer.addTrace(category, requestType);
+      if (isInvalidMocker(requestType.getRecordId())) {
+          LOGGER.warn("recordId: {} is invalid", requestType.getRecordId());
+          return ResponseUtils.parameterInvalidResponse("invalid mocker");
+      }
 
       boolean saveResult = agentWorkingMetricService.saveRecord(requestType);
       LOGGER.info("agent record save result:{},category:{},recordId:{}", saveResult, category,
           requestType.getRecordId());
       return ResponseUtils.successResponse(saveResult);
-    } catch (Throwable throwable) {
-      LOGGER.error("save record error:{} from category:{},recordId:{}", throwable.getMessage(),
-          category,
-          requestType.getRecordId(), throwable);
-      return ResponseUtils.exceptionResponse(throwable.getMessage());
+    } catch (Exception e) {
+      LOGGER.error("save record error: {} from category: {}, recordId: {}",
+          e.getMessage(), requestType.getCategoryType(), requestType.getRecordId(), e);
+      handleSaveMockerError(requestType);
+      return ResponseUtils.exceptionResponse(e.getMessage());
     } finally {
       MDCTracer.clear();
     }
   }
 
+  @PostMapping(value = "/batchSave")
+  @ResponseBody
+  public Response batchSave(@RequestBody byte[] request) {
+    if (request == null || request.length == 0) {
+        return ResponseUtils.parameterInvalidResponse("request body is empty");
+    }
+
+    try {
+      List<AREXMocker> mockers = zstdJacksonSerializer.deserialize(request, new TypeReference<List<AREXMocker>>() {});
+      if (CollectionUtils.isEmpty(mockers)) {
+        return ResponseUtils.parameterInvalidResponse("No valid data found in the request body");
+      }
+
+      if (isInvalidMocker(mockers.get(0).getRecordId())) {
+        return ResponseUtils.parameterInvalidResponse("invalid mocker");
+      }
+
+      CompletableFuture.supplyAsync(() -> batchSaveMocker(mockers), customForkJoinExecutor);
```

Comment:
how many coreSize of customForkJoinExecutor?   
will there be thread accumulation issues in high concurrency situations?

Reply:
In the current scenario, the time consumption of a single save interface is 2ms, the qps is around 400, and this interface is an IO-intensive interface, so the number of core threads is set to 200